### ISSUE TITLE
Porter doesn't seem to have any options for configuration behind a corporate proxy?

### ISSUE THREAD

**Issue by HarveyEllis** [getporter/porter#927]
## Describe the bug
Dockerfile doesn't build behind corporate proxy.

## To Reproduce
I am using MacOS 10.15 Catalina.

1. Set proxy configuration in docker desktop
2. Run `porter build` behind a corporate proxy

## Expected behavior
I have been testing Porter at a large company, and can't see how I can get porter to use to the correct proxy settings for building the docker image. Whilst we can `squidman` out of the proxy, this is generally not recommended.

Usually when building a docker image I have to inject proxy args using `--build-arg` and also have to copy our `ca-cert.crt` file across and set curl/wget to use it. 

There should be some way to inject proxy configuration and certificate bundles into the dockerfile.

## Porter Command and Output
```
porter build
```
Specific output:
```
Error: unable to build CNAB invocation image: failed to stream docker build output: The command '/bin/sh -c curl "https://s3.amazonaws.com/aws-cli/awscli-bundle.zip" -o "/tmp/awscli-bundle.zip"' returned a non-zero code: 60
```
Basically the docker daemon doesn't seem to pick up on proxies. There are both errors around accessing the debian package repositories, and also errors relating to SSL certs.

## Version
All versions tried. Current version tried: porter v0.23.0-beta.1 (9c49056)

**Comment by carolynvs** [comment]
@HarveyEllis Thanks for the report!

Quick question, are you using a [custom dockerfile](https://porter.sh/custom-dockerfile/) right now? I am wondering if as a workaround for now, you can use that and hard-code the build argument that you need and add the extra COPY statements to get curl/wget and anything else necessary for your proxy. 

If that doesn't work for you, don't worry, I would like to support a proxy, just throwing out ideas to help you get by until then though I would like to understand why not so I don't go suggesting stuff like this to people. 😀 

Longer term, porter should detect that docker has been configured to use a proxy and just magically use it too, without additional futzing.

**Comment by HarveyEllis** [comment]
@carolynvs thanks very much for the reply. I have looked at using the custom dockerfile - and it does provide the workaround needed. 

I just wanted to raise this as an issue, as I looked through the docs and couldn't seem to find anything on configuring proxies/custom SSL certs. Tbh, perhaps the custom dockerfile is the best way given the complexity of some corporate setups.

Thanks again! 😄

**Comment by carolynvs** [comment]
Another related issue is #1941
